### PR TITLE
fix: ito claim re-check

### PR DESCRIPTION
**Pull request by zhouhanseng**

### CODE REVIEW

File: packages/maskbook/src/plugins/ITO/hooks/useClaimCallback.ts [hunk 1]
```diff
@@ -33,7 +34,35 @@ export function useClaimCallback(pids: string[]) {
...
+            const isClaimed = (
+                await Promise.all(
+                    pids.map(async (pid) => {
+                        const availability = await ITO_Contract.methods.check_availability(pid).call({
+                            from: account,
+                        })
+                        return new BigNumber(availability.swapped).isZero()
+                    }),
+                )
+            ).some((claimed) => claimed)
```
Comment (guanbinrui) [inline packages/maskbook/src/plugins/ITO/hooks/useClaimCallback.ts:48]: Boolean

Comment (guanbinrui) [inline packages/maskbook/src/plugins/ITO/hooks/useClaimCallback.ts:48]:
```suggestion
            ).some(Boolean)
```

**Review by guanbinrui — DISMISSED** [review]

File: packages/maskbook/src/plugins/ITO/hooks/useClaimCallback.ts [hunk 2]
```diff
@@ -33,7 +34,35 @@ export function useClaimCallback(pids: string[]) {
...
+        try {
+            const isClaimed = (
+                await Promise.all(
+                    pids.map(async (pid) => {
+                        const availability = await ITO_Contract.methods.check_availability(pid).call({
+                            from: account,
+                        })
+                        return new BigNumber(availability.swapped).isZero()
+                    }),
+                )
```
Comment (septs) [inline packages/maskbook/src/plugins/ITO/hooks/useClaimCallback.ts:47]: extract as variable.

File: packages/maskbook/src/plugins/ITO/hooks/useClaimCallback.ts [hunk 3]
```diff
@@ -33,7 +34,35 @@ export function useClaimCallback(pids: string[]) {
...
 
-        // step 1: estimate gas
+        // step 1: check if already claimed
+        try {
+            const availabilityList = await Promise.all(
+                pids.map(async (pid) => {
+                    const availability = await ITO_Contract.methods.check_availability(pid).call({
+                        from: account,
+                    })
+                    return availability
```
Comment (septs) [inline packages/maskbook/src/plugins/ITO/hooks/useClaimCallback.ts:44]:
```suggestion
                    return ITO_Contract.methods.check_availability(pid).call({
                        from: account,
                    })
```

File: packages/maskbook/src/plugins/ITO/hooks/useClaimCallback.ts [hunk 4]
```diff
@@ -33,7 +34,35 @@ export function useClaimCallback(pids: string[]) {
...
+                }),
```
Comment (septs) [inline packages/maskbook/src/plugins/ITO/hooks/useClaimCallback.ts:45]:
```suggestion
                pids.map((pid) => ITO_Contract.methods.check_availability(pid).call({
                    from: account,
                })),
```

**Review by guanbinrui — APPROVED** [review]

**Review by septs — APPROVED** [review]

**Pull request merged**